### DISCUSSION TITLE
Prototyping a Faster Optuna Implementation in Rust

### DISCUSSION THREAD

**Discussion by HideakiImamura** [optuna/optuna#5362] [Announcements]
We’ve started prototyping the faster Optuna implementation in Rust, featuring Python, JavaScript, and C bindings.

Masashi Shibata (@c-bata) explains the motivation and the potential benefits that a Rust-based Optuna implementation could provide. 🚀

[https://medium.com/optuna/prototyping-a-faster-optuna-implementation-in-rust-e76efba3761b](https://t.co/dkrhpIT4hF)

**Comment by alexisrosuel** [comment]


**Reply by nzw0301** [reply]
Sorry it was a private repo, never mind.

**Comment by jmaczan** [comment]
Hi, do you still work on the prototype or did you decide to drop it?

**Comment by mpasa** [comment]
Is there any plans to recover efforts on the Rust implementation? I would be very interested in using Optuna from Rust, or sharing trials using Pyo3 bindings.

**Comment by stastnypremysl** [comment]
Hi, @HideakiImamura. What is the current status please?

**Reply by mpasa** [reply]
@stastnypremysl Hideaki wrote about the Optuna v5 roadmap [here](https://medium.com/optuna/optuna-v5-roadmap-ac7d6935a878), which includes quite some references to Rust and an expected release date.

**Reply by stastnypremysl** [reply]
Thanks.

**Comment by c-bata** [comment]
@stastnypremysl @mpasa @jmaczan @alexisrosuel Thanks for your interest, and sorry for the long silence. The Rust implementation is now publicly available!

* GitHub: https://github.com/optuna/rustuna/
* Blog post: https://medium.com/optuna/announcing-rustuna-cc82a6815bf7
